conmon/conmon.c

```diff
@@ -825,13 +851,16 @@ static gboolean ctrl_cb(int fd, G_GNUC_UNUSED GIOCondition condition, G_GNUC_UNU
 			return G_SOURCE_CONTINUE;
 		}
 		ninfo("Message type: %d, Height: %d, Width: %d", ctl_msg_type, height, width);
-		ret = ioctl(masterfd_stdout, TIOCGWINSZ, &ws);
-		ninfo("Existing size: %d %d", ws.ws_row, ws.ws_col);
-		ws.ws_row = height;
-		ws.ws_col = width;
-		ret = ioctl(masterfd_stdout, TIOCSWINSZ, &ws);
-		if (ret == -1) {
-			nwarn("Failed to set process pty terminal size");
+		switch (ctl_msg_type) {
+			// This matches what we write from container_attach.go
+			case 1:
+				resize_winsz(height, width);
+				break;
+			case 2:
+				reopen_log_file();
```

Comment:
Won't we leak the already open `log_fd` here if we don't close it first?

Reply:
:+1: I'll fix it.

Reply:
Moved the closing of log_fd to the reopen function.